src/idiomorph.js

```diff
@@ -1063,11 +1074,49 @@ var Idiomorph = (function () {
         function removeNode(tempNode, ctx) {
             removeIdsFromConsideration(ctx, tempNode)
             if (ctx.callbacks.beforeNodeRemoved(tempNode) === false) return;
-
-            tempNode.parentNode?.removeChild(tempNode);
+            if (ctx.pantry) {
+                moveToPantry(tempNode.pantry, ctx);
+            } else {
+                tempNode.parentNode?.removeChild(tempNode);
+            }
             ctx.callbacks.afterNodeRemoved(tempNode);
         }
 
+        function moveToPantry(node, ctx) {
+            if (!node) return;
+
+            // If the node is a leaf (no children), process it, and then we're done
+            if (!node.children || node.children.length === 0) {
+                if (node.id) {
+                    ctx.pantry.appendChild(node);
+                }
+
+            // otherwise we need to process the children first
+            } else {
+                Array.from(node.children).forEach(child => {
+                    moveToPantry(child, ctx);
+                });
+
+                // After processing children, process the current node
+                if (node.id) {
+                    ctx.pantry.appendChild(node);
+                }
+            }
+        }
+
+        function restoreFromPantry(root, ctx) {
+            Array.from(ctx.pantry.children).forEach(element => {
+                const matchElement = root.findElementById(element.id);
+                if (matchElement) {
+                    matchElement.before(element);
```

Comment:
looks reasonable, want to also have a `moveBefore()` branch

maybe we empty the children out of the old node when it is stuck in the pantry?